WalletWasabi/WabiSabi/Client/RoundStateUpdater.cs

```diff
@@ -75,6 +86,9 @@ public override Task StopAsync(CancellationToken cancellationToken)
 					awaiter.Cancel();
 				}
 			}
+
+			_personCircuit?.Dispose();
```

Comment:
it seems to me that this is done just to hack the design because if one component is disposable then all the other components that own one instance of that component have to be also disposable. So, the request handlers (at least one particular implementation) needs to own a `PersonCircuit` (disposable) and then that request handler has to be make disposable. But instead of doing that the concrete low-level dependency is passed and disposed directly.

Imo we have to keep the design with the abstract dependencies and the clean design and make the request handler disposable and make sure it disposes the owned personcircuit.

Reply:
This is indeed a hack. I have reworked it yesterday locally but there is still somewhere a bug. I'll push later.